src/spatial/traversal/banking/MemoryConfigurer.scala

```diff
@@ -111,28 +111,70 @@ class MemoryConfigurer[+C[_]](mem: Mem[_,C], strategy: BankingStrategy)(implicit
     }
   }
 
+  /** True if a and b always occur at the exact same time.
+    * This is trivially true if a and b are the same unrolled access.
+    *
+    * This method is used to enable broadcast reads.
+    */
+  def canBroadcast(a: AccessMatrix, b: AccessMatrix): Boolean = {
+    // TODO[3]: What about accesses of the same form across different loops?
+    // Should we rely on loop fusion for this? Are there cases where that wouldn't work?
+    val isWrite = a.access.isWriter || b.access.isWriter
+    if (isWrite || a.access != b.access || a.matrix != b.matrix) return false
+
+    val iters = accessIterators(a.access, mem)
+    // The index of iterators which will differ between a and b
```

Comment:
Could the following still be an issue here?  It's already listed as issue #55 so the fix to that will fix this as well.

The iters will look the same for j before unrolling but j actually holds different values in lock-step since i changes per unrolled copy of the j loop.

```
Foreach(N by 1 par p){i => 
  Foreach(i until N by 1){j => 
    a(i,j) = b(j) * b(i)
  }
)